Adding more retriever examples to the documentation examples (#116196)
elastic/elasticsearch: docs/reference/search/search-your-data/retrievers-examples.asciidoc

[[retrievers-examples]]
=== Retrievers examples

> 

Reviewer: ```suggestion

Learn how to combine different retrievers in these hands-on examples.
```

To demonstrate the full functionality of retrievers, these examples require access to a <<semantic-reranking-models,semantic reranking model>> set up using the <<inference-apis,Elastic inference APIs>>.

> To begin with, we'll set up the necessary services and have them in place for later use.

Reviewer: ```suggestion
[discrete]
[[retrievers-examples-setup]]
==== Add example data

To begin with, we'll set up the necessary services and have them in place for later use.
```

> [source,js]

Reviewer: As much as possible we should use [source, console] and [test snippets](https://github.com/elastic/elasticsearch/tree/main/docs#test-code-snippets). 

See this [example doc](https://github.com/elastic/elasticsearch/blob/main/docs/reference/search/retriever.asciidoc?plain=1).

You'll need to skip stuff that requires ML or semantic reranking, but everything else should be tested.

Reviewer: This could be a follow up PR to not block but one of the big strengths of the ES docs is that they're tested :)

Author: Yeap, that's fair! Will follow up with another PR :)

 "year": 2023,
 "topic": ["ai", "medicine"]
}

POST /retrievers_example/_doc/3
{
 "vector": [0.45, 0.32, 0.91],
  "text": "AI is redefining security by enabling advanced threat detection, proactive risk analysis, and dynamic defenses against increasingly sophisticated cyber threats.",
 "year": 2024,
 "topic": ["ai", "security"]
}

POST /retrievers_example/_doc/4
{
 "vector": [0.34, 0.21, 0.98],
 "text": "Elastic introduces Elastic AI Assistant, the open, generative AI sidekick powered by ESRE to democratize cybersecurity and enable users of every skill level.",
 "year": 2023,
 "topic": ["ai", "elastic", "assistant"]
}

POST /retrievers_example/_doc/5
{
 "vector": [0.11, 0.65, 0.47],
 "text": "Learn how to spin up a deployment of our hosted Elasticsearch Service and use Elastic Observability to gain deeper insight into the behavior of your applications and systems.",
 "year": 2024,
 "topic": ["documentation", "observability", "elastic"]
}

----
//NOTCONSOLE

Now that we also have our documents in place, let's try to run some queries using retrievers.

[discrete]
[[retrievers-examples-combining-standard-knn-retrievers-with-rrf]]

> ==== Combining query and kNN with RRF

Reviewer: ```suggestion
==== Example: Combining query and kNN with RRF
```

@pmpailis it now actually makes more sense in the TOC to help separate the different examples

So can replicate this for each subheading on this page :)

[[retrievers-examples]]
=== Retrievers examples

> To show the full functionality, for this set of examples, we'll assume that we have access to a <<semantic-reranking-models,reranker model>> through the inference service.

Reviewer: ```suggestion
To demonstrate the full functionality of retrievers, these examples require access to a <<semantic-reranking-models,reranker model>> set up using the <<inference-apis,Elastic inference APIs>>.
```
